[Delton Cheng Xu Zhi] iP

DeltonCheng opened the pull request:
# Chadbot
> “help me pls” – me ([source](https://www.youtube.com/watch?v=dQw4w9WgXcQ&ab_channel=RickAstley))

Chadbot is a bot to help you remember your tasks!

Chadbot is:

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/DeltonCheng/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)


If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

DeltonCheng committed:
test 1 commit level-0

DeltonCheng committed:
Level-1 Echo

DeltonCheng committed:
Level-2

DeltonCheng committed:
Level-3 Mark as Done

DeltonCheng committed:
Level-4 ToDos, Events, Deadlines

DeltonCheng committed:
Level-5 Handle Errors

DeltonCheng committed:
Level-6 Delete

DeltonCheng committed:
A-TextUiTesting

DeltonCheng committed:
Level-7 Save

DeltonCheng committed:
Level-8 Date and Times

DeltonCheng committed:
A-More OOP

DeltonCheng committed:
A-More OOP
- some cleaning

DeltonCheng committed:
A-Packages

DeltonCheng committed:
Merge commit 'ed6d4d25bd4e34461919ffe785a6f10f4c087552'

DeltonCheng committed:
A-JUnit

DeltonCheng committed:
A-Jar

DeltonCheng committed:
A-JavaDoc
Completed for Ui

DeltonCheng committed:
A-JavaDoc
Completed for the rest of classes

DeltonCheng committed:
A-JavaDoc
Completed for the rest of classes

DeltonCheng committed:
A-CodingStandard
done for all classes

DeltonCheng committed:
Level-9

DeltonCheng committed:
Merge branch 'A-CodingStandard'

DeltonCheng committed:
Level-9
Resolve conflict

DeltonCheng committed:
Level-9
Cleaning

DeltonCheng committed:
Level- 9
Resolved conflicts part 2

jianrong7 commented on README.md:
Nice to see that you have updated the README. :)

jianrong7 commented on src/main/java/actions/TaskList.java:
```diff
@@ -0,0 +1,98 @@
+
+public class TaskList {
+    private ArrayList<Task> taskArrayList;
+    String line = "~~*~~*~~*~~*~~*~~*~~*~~*~~*~~\n";
```
This line seems to be a constant, perhaps change it to SCREAMING_SNAKE_CASE. Also, do add access modifiers to it.
perhaps
`private final String LINE = "~~*~~*~~*~~*~~*~~*~~*~~*~~*~~\n";`

jianrong7 commented on src/main/java/actions/TaskList.java:
```diff
@@ -0,0 +1,98 @@
+     * @param index of task to be marked
+     */
+    public void chadMarkTask(int index) {
+        taskArrayList.get(index - 1).isComplete = true;
```
Not sure about directly accessing the `isComplete` field of your task. It should be private and should be accessed through a setter.

jianrong7 commented on src/main/java/actions/TaskList.java:
```diff
@@ -0,0 +1,98 @@
+     * @param index of task to be unmarked
+     */
+    public void chadUnmarkTask(int index) {
+        taskArrayList.get(index - 1).isComplete = false;
```
Not sure about directly accessing the `isComplete` field of your task. It should be private and should be accessed through a setter.

jianrong7 commented on src/main/java/actions/TaskList.java:
```diff
@@ -0,0 +1,98 @@
+            }
+        }
+        if (matched.isEmpty()) {
+            System.out.println(line);
+            System.out.println("There are no matching tasks!");
+            System.out.println(line);
+        } else {
+            System.out.println(line);
+            System.out.println("Here are the matching tasks:");
+            for (int i = 0; i < matched.size(); i ++) {
+                System.out.println(matched.get(i).toString());
+            }
+            System.out.println("Total of " + matched.size() + " matched tasks");
+            System.out.println(line);
```
perhaps let `Ui.java` handle these lines?

jianrong7 commented on src/main/java/main/Duke.java:
```diff
@@ -0,0 +1,62 @@
+
+    private Ui ui;
+    private Storage storage;
+    private TaskList tasklist;
+    public ArrayList<Task> taskArrayList = new ArrayList<>();
```
I think `taskList` should already be declared in `taskArrayList` and not be accessible by `Duke` here.
Why is the access modifier for `taskArrayList` public?

jianrong7 commented on src/main/java/main/Duke.java:
```diff
@@ -0,0 +1,62 @@
+        scanObj.close();
+    }
+
+    public static void main (String[]args){
```
Be careful of spacing here.
Should be `public static void main(String[] args) {`

jianrong7 commented on src/main/java/tasks/Deadline.java:
```diff
@@ -0,0 +1,36 @@
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+    public LocalDate date;
```
Why is the access modifier public?

jianrong7 commented on src/main/java/tasks/Event.java:
```diff
@@ -0,0 +1,41 @@
+
+public class Event extends Task {
+
+    public LocalDate dateFrom;
+    public LocalDate dateTo;
```
Perhaps change this to `private` and make use of getters.

jianrong7 commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,42 @@
+package tasks;
+
+public class Task {
+    public String name;
+    public boolean isComplete;
```
Should be `private` and access them through getters and setters.

jianrong7 commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,111 @@
+package ui;
+
+
+public class Ui {
+    String line = "~~*~~*~~*~~*~~*~~*~~*~~*~~*~~\n";
```
Already declared above in TaskList.java. Should be reused there. Also should be a constant, so SCREAMING_SNAKE_CASE should be used.

jianrong7 reviewed (commented):
I like your code and it is generally good. Just some things to take note about access modifiers. Good job! 😸

DeltonCheng committed:
Level-10
Completed up till tutorial 3

DeltonCheng committed:
Level-10
Tutorial 4

DeltonCheng committed:
Level-10
Linked GUI to chadbot

Leb14 commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,135 @@
+
+        boolean mark = marked.equals("X");
+
+        switch (type) {
```
It's nice to use case here to improve code readability!

Leb14 commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,42 @@
+     *
+     * @return "X" if task completed and an empty string if not completed
+     */
+    public String getMark() {
```
Will it be better if you let the parser do this?

Leb14 commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,267 @@
+            return false;
+
+        } else if (inputArray[0].equals("list")) {
+            ui.chadListTask(chad.taskArrayList);
```
I think it's better to create a Command package to handle all the commands.

DeltonCheng committed:
A-Assertion

DeltonCheng committed:
A-CodeQuality
Cleaned up some code

DeltonCheng committed:
A-Assertion

DeltonCheng committed:
A-CodeQuality
Added more javadocs

DeltonCheng committed:
Merge branch 'A-CodeQuality'

DeltonCheng committed:
BCD-Extension

DeltonCheng committed:
BCD-Extension

DeltonCheng committed:
A-BetterGui

DeltonCheng committed:
A-Personality
Added Javadocs

DeltonCheng committed:
A-Personality
Added Javadocs

DeltonCheng committed:
A-BetterGui

GUI only has basic features, lack any originality

To differentiate the project's GUI from others, practice using JavaFX

Added colours and borders around certain aspects to make it stand out

Let us ensure that the UX remain user-friendly while improving uniqueness

DeltonCheng committed:
Merge pull request #2 from DeltonCheng/A-BetterGui

A-BetterGui

DeltonCheng committed:
A-CodeQuality2

Parser method is too long

To improve readability by abstracting commands out

Added commands and added them to parser

Let us ensure readability by reducing deep nesting / arrowheads

DeltonCheng committed:
A-CodeQuality2

Parser method is too long and unused TextUI parser

To improve readability by abstracting commands out and remove TextUI parser

Added the rest of commands and added them to parser

Let us ensure readability by reducing deep nesting / arrowheads

DeltonCheng committed:
Updated README

DeltonCheng committed:
Minor code cleaning

DeltonCheng committed:
Minor code cleaning

DeltonCheng committed:
Update README again

DeltonCheng committed:
Added find to README

DeltonCheng committed:
Debugged and allowed for cross platform
